### PR TITLE
Exams by Bogdan Lepkov, IM-23

### PR DESCRIPTION
**Pull request by nevrodegenerate**

### CODE REVIEW

File: Tasks/between.js [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-      i = str.indexOf(s);
-      if (i === -1) {
-        return '';
-      } else {
-        str = str.substring(0, i);
-      }
-    }
-  }
-  return str;
+const getvaluebetween = (fullString, startString, endString) => {
```
Comment (v-jester) [inline Tasks/between.js:4]:
```suggestion
const getValueBetween = (fullString, startString, endString) => {
```
code style not visible

File: Tasks/month.js [hunk 2]
```diff
@@ -15,10 +16,9 @@ Months = [
   'dec',
 ];
 
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+const Month = (inputString) => {
```
Comment (tshemsedinov) [inline Tasks/month.js:19]: Use `camelCase`

File: Tasks/password.js [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-  const MAX = alphabet.length;
-  key = '';
-  for (let i = 0; i < length; i++) {
-    Index = Math.floor(Math.random() * MAX);
-    key = key + alphabet[Index];
-  }
-  return key;
+const GeneratePassword = (alphabet, passLength) => {
+  let password = '';
+  for (let i = 0; i < passLength; i++) password += alphabet[Math.floor(Math.random() * alphabet.length)];
```
Comment (tshemsedinov) [inline Tasks/password.js:6]: Use `{}` blocks

Comment (tshemsedinov) [reply Tasks/password.js:6]: I'd prefer to extract `random` to a separate function

File: Tasks/quotes.js [hunk 4]
```diff
@@ -1,30 +1,23 @@
...
-  open = false;
-  for (c of s) {
-    if (c === '"') {
-      for (i of c) {
-        if (!open) {
-          res.push('«');
-          open = true;
-        } else {
-          res.push('»');
+const quotes = function (inputString) {
```
Comment (tshemsedinov) [inline Tasks/quotes.js:4]: Prefer arrow-functions
```suggestion
const quotes = (inputString) => {
```

File: Tasks/quotes.js [hunk 5]
```diff
@@ -1,30 +1,23 @@
...
+  let outputString = '';
+  let open = false;
+  for (const SYMBOL of inputString) {
```
Comment (tshemsedinov) [inline Tasks/quotes.js:7]:
```suggestion
  for (const char of inputString) {
```

File: Tasks/quotes.js [hunk 6]
```diff
@@ -1,30 +1,23 @@
...
+    if (SYMBOL === '"') {
+        if (open) {
+          outputString += '»';
           open = false;
+        } 
+        else {
+          outputString += '«';
```
Comment (tshemsedinov) [inline Tasks/quotes.js:14]: Use ternary for open/close quotes

File: Tasks/quotes.js [hunk 7]
```diff
@@ -1,30 +1,23 @@
...
+          open = true;
         }
```
Comment (tshemsedinov) [inline Tasks/quotes.js:16]: `open = !open`

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Code style and formatting is terrible
- Use eslint and prettier before commits
- You need to view almast all lectures again and try to implement all tasks for practice

**Pull request closed without merging**